### DISCUSSION TITLE
Selective Text on Stage Display

### DISCUSSION THREAD

**Discussion by Hysterelius** [ChurchApps/ChurchAppsSupport#436] [FreeShow] [answered]
Hi, I would love to be able to only show text on a stage display. Currently, I have been doing it by setting the main text to have an opacity of zero through css - but that means I cannot see the text on the main show "management" screen when doing the presentation.

I also have a live output, which shows text as a banner on the bottom of the screen and I would like to hide this banner when there is no text (as I am using a custom style).

Any help would be appreciated :)

**Comment by vassbo** [comment, marked as answer]
You can right click any textbox item, and choose "Specific outputs">"Stage"

**Reply by Hysterelius** [reply]
That worked for me, thanks :)

![image](https://github.com/user-attachments/assets/76f9177e-90aa-41fc-b5c8-1b02ac8a6a93)

However, I was wondering how to get it not show on my live display - and if there was a way to hide the grey text box (I changed the background colour to white for clarity)
_the live display is the top left one_

These are the settings for live:
![image](https://github.com/user-attachments/assets/0b160675-cc99-4a41-9445-9cdef39e3970)

**Reply by vassbo** [reply]
It should not show up at all in the "Live" output if you have the "Specific outputs" set to just "Stage" and not "Live" for that textbox item?!

**Reply by Hysterelius** [reply]
Yep, I just selected "Stage" for the textbox - (I have made sure to reload the output)

![image](https://github.com/user-attachments/assets/1d5a0180-96a8-4d56-95a6-3a7587e0a1ae)


I think it may be from the live template overriding it

**Reply by vassbo** [reply]
I see, try changing the template for "Live" as well.

**Reply by Hysterelius** [reply]
What should I change it to?

This is the template, it is pretty basic:

![image](https://github.com/user-attachments/assets/afc84515-6943-4b24-a48c-2b25266c1255)

And it is applied in these settings:

![image](https://github.com/user-attachments/assets/58ac4c54-5248-4eb4-9f0e-f7ffa0d75941)

**Reply by vassbo** [reply]
Right click the textbox here and set the "Specific outputs" to "Stage" here as well.

**Reply by Hysterelius** [reply]
By textbox here do you mean the one on the "Live" style?

I tried that, and that just made the live style blank

**Reply by Hysterelius** [reply]
I found if I do the specific output as just "Live" on the original textbox it does nothing, while if I make it "Primary" it hides it from the stage display

![image](https://github.com/user-attachments/assets/eb85656d-51fb-4486-b372-1ce620db729c)

**Reply by vassbo** [reply]
That template yes, but of course if you just want this on just a certain show, then you need to create an action to remove the output style for the "Live" output when the first slide in that specific show is played, and then one to re-add it on the last slide.

**Reply by Hysterelius** [reply]
If this is a bug I can try fix this, probably need some help though... not used to Svelte

**Reply by vassbo** [reply]
It's not a bug. It's because of the template in your output style.

**Reply by Hysterelius** [reply]
Shouldn't the template respect that it shouldn't be displayed on Live?

The template is just a variation in text position though.

**Reply by vassbo** [reply]
Just tested it, and it works as it should. The textbox from the template should and does override the textbox on the slide.

At first I set a custom template on the "Live" output.
At first I set the textbox specific outputs to just "Stage".

Now when I display it shows on the Stage and in the Live output, not Primary.
So I also set the template textbox specific outputs to "Stage" (something other than "Live").
And now when I present, the textbox only shows on stage as it should!

Now if I want it to only show on stage in certain scenarios, then I can simply remove the output style with the custom template instead of setting the specific output style on that slide. And that also makes it just show up in stage. (As I mentioned you can use actions to automate that!)
